src/main/java/com/google/edith/servlets/ReceiptData.java

```diff
@@ -0,0 +1,95 @@
+package com.google.edith.servlets;
+
+import com.google.appengine.api.users.User;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.cloud.documentai.v1beta2.DocumentUnderstandingServiceClient;
+import com.google.edith.interfaces.ExtractReceiptInterface;
+import com.google.edith.services.ExtractReceiptService;
+import java.io.IOException;
+import java.util.List;
+import java.util.Map;
+
+/** Converts parsed receipt file string to receipt and item objects. */
+public final class ReceiptData {
+
+  private ExtractReceiptInterface extractReceiptImplementation;
+
+  public ReceiptData() throws IOException {
+    this.extractReceiptImplementation =
+        new ExtractReceiptService(DocumentUnderstandingServiceClient.create());
+  }
+
+  public ReceiptData(ExtractReceiptInterface extractReceiptImplementation) {
+    this.extractReceiptImplementation = extractReceiptImplementation;
+  }
+
+  /**
+   * Creates receipt object.
+   *
+   * @param blobKey - string representation of the blob in GCS bucket.
+   * @param expenditureName - name of the rceipt obtained from FE form.
+   * @return Receipt - receipt object created from the provided info.
+   */
+  public Receipt extractReceiptData(String blobKey, String expenditureName) throws IOException {
+
+    UserService userService = UserServiceFactory.getUserService();
```

Comment:
This looks like a dependency that should be initialized in the constructor to allow for easier testing and control on the dependency to the caller

Reply:
Removed the dependency to service class.